src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -240,8 +263,13 @@ bool meetPoolParamRestrictions(int64_t inputShape, int64_t kernelShape,
 }
 
 /// Default legality check.
+/// This template is not directly called.
 template <typename OP_TYPE>
 bool isSuitableForZDNN(OP_TYPE op, const DimAnalysis *dimAnalysis) {
+  // Check NNPA level.
+  if (!isCompatibleWithNNPALevel(NNPA_Z16)) {
+    return false;
+  }
```

Comment:
Looks redundant. Just returning `false` for unsupported ops is enough.

Reply:
Thanks, I updated to just return false.